### PR TITLE
Add device twin e2e tests and JsonSerializerSettingsInitializer setups with unit tests for DPS clients

### PR DESCRIPTION
**Pull request by brycewang-microsoft**

### CODE REVIEW

File: e2e/test/iothub/device/TwinE2ETests.cs [hunk 1]
```diff
@@ -294,6 +299,42 @@ public async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesItOnNextGet_Am
                 .ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesItOnNextGet_DateTimeProperties_Mqtt()
```
Comment (abhipsaMisra) [inline e2e/test/iothub/device/TwinE2ETests.cs:304]: The serailization behavior is independent of the transport protocol used, so you should be ok adding test for any one variant and removing the rest. Add in a comment that calls this out and it should be good.

**Review by abhipsaMisra — APPROVED** [review]

File: e2e/test/iothub/device/TwinE2ETests.cs [hunk 2]
```diff
@@ -691,17 +732,37 @@ private async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesItOnNextGetAs
...
             bool propertyFound = deviceTwin.Desired.TryGetValue(propName, out string actual);
             propertyFound.Should().BeTrue();
             actual.Should().Be(propValue);
 
             await deviceClient.CloseAsync().ConfigureAwait(false);
         }
 
+        private async Task Twin_ServiceSetsDesiredPropertyAndDeviceReceivesItOnNextGetDateTimePropertiesAsync(IotHubClientTransportSettings transportSettings)
+        {
+            string propName = "Iso8601String";
```
Comment (schoims) [inline e2e/test/iothub/device/TwinE2ETests.cs:748]: nit: 
```suggestion
            const string propName = "Iso8601String";

**Pull request merged**